src/Schema/Provider/PhpFileSchemaProvider.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Cycle\Schema\Provider;
+
+use Cycle\ORM\Schema;
+use RuntimeException;
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\Yii\Cycle\Schema\Converter\SchemaToPHP;
+use Yiisoft\Yii\Cycle\Schema\SchemaProviderInterface;
+
+/**
+ * Be careful, using this class may be insecure.
+ */
+final class PhpFileSchemaProvider implements SchemaProviderInterface
+{
+    public const MODE_READ_AND_WRITE = 0;
+    public const MODE_WRITE_ONLY = 1;
+
+    private string $file = '';
+    private int $mode = self::MODE_READ_AND_WRITE;
+
+    private Aliases $aliases;
+
+    public function __construct(Aliases $aliases)
+    {
+        $this->aliases = $aliases;
+    }
+
+    public function withConfig(array $config): self
```

Comment:
1. How about having two separate methods for file and mode?
2. Looks like file is mandatory for the class. Is it better to make it part of the constructor?

Reply:
Following the current [configuration mechanism](https://github.com/yiisoft/yii-demo/blob/b0b91a45d1726094548a6547c697a06c567568a5/config/params.php#L73-L83), it is not advisable to transfer a parameter to a constructor. There is no need for an additional method, but it can be added